pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -619,7 +656,7 @@ def _check_function(self, node: nodes.FunctionDef, match: TypeHintMatch) -> None
                 )
 
         # Check the return type.
-        if not _is_valid_type(return_type := match.return_type, node.returns):
+        if not _is_valid_return_type(return_type := match.return_type, node.returns):
```

Comment:
Maybe it's worth passing the whole `match` object to `_is_valid_return_type`. Especially if there is another attribute for the inheritance check.